geonode/catalogue/views.py

```diff
@@ -108,9 +108,13 @@ def csw_global_dispatch(request):
                 for group in request.user.groups.all():
                     groups_ids.append(group.id)
 
-            public_groups = GroupProfile.objects.exclude(access="private").values('group')
+            public_groups = GroupProfile.objects.exclude(access="private").values('group').distinct()
             for group in public_groups:
```

Comment:
Which is the case when we have more than one group per GroupProfile?

Reply:
no case, you are right. It is a refuse. Removing. Thanks @simod